### ISSUE TITLE
[UX] CKEditor - Provide default styles 

### ISSUE THREAD

**Issue by stpaultim** [backdrop/backdrop-issues#4233]
**Description of the need**
Make it easier for new users to figure out how to edit the styles menu option in CKEditor. 

I just spent a few hours trying to figure out how to get this to work. In the end, my problem was silly human error in that I was trying to get an h1 style to work while h1 were not listed as part of the allowed HTML tags for that text format. But, having default styles available at the start, would have made it much easier for me to see that they were working and to have debugged my problem. Having no styles available left me with the impression that the feature might be broken and discouraged me from even trying to find the problem. 

Not having any default styles to start with seems very counter-intuitive to me. 

Here is an issue I opened while trying to figure this out: https://github.com/backdrop/backdrop-issues/issues/4232

**Proposed solution**
Provide some default `styles` in `styles list` for the Filtered HTML text format.  

These could be added in the install profile to make the `styles` menu useful out of the box, but mostly to demonstrate to new users how it works and make it easier for people to figure out how to edit them. 

Just need to figure out what some sensible styles would be to provide by default. Keep in mind, that they styles menu is not even available by default. So these default styles would only be used or even visible if a user tried to add this option to their WYSIWYG menu. 

**Alternatives that have been considered**
Related issues (some of which might mitigate need for this):
https://github.com/backdrop/backdrop-issues/issues/1520
https://github.com/backdrop/backdrop-issues/issues/2980

An alternative would be to include a "No styles configured" message under the styles menu in the WYSIWYG if the styles menu is enabled, but not configured. 

**Additional information**

This is what you see in the WYSIWYG after adding the styles option to the toolbar (but before adding any custom styles).

![Create_Page___Styles](https://user-images.githubusercontent.com/3144571/70399990-c322c200-19ed-11ea-8d7c-cb42826aa0cf.png)

Once you click on it, you get an empty box - this feels very broken.

![Create_Page___Styles](https://user-images.githubusercontent.com/3144571/70400016-e64d7180-19ed-11ea-8030-61dd92ae148d.png)

**Comment by olafgrabienski** [comment]
> Related issues (some of which might mitigate need for this):
> #1520
> #2980

In https://github.com/backdrop/backdrop-issues/issues/2980#issuecomment-508772952 @docwilmot suggested to add tags from the Styles list automatically to the Allowed HTML tags. I guess, that would have been helpful in https://github.com/backdrop/backdrop-issues/issues/4232. For #2980, there is even a PR by @quicksketch that should add the tags but it needs work.

**Comment by olafgrabienski** [comment]
> Just need to figure out what some sensible styles would be to provide by default.

Here are some candidates from https://github.com/backdrop/backdrop-issues/issues/2980#issuecomment-366750627:

> Besides "small" tags, there are a number of other inline tags that CKEditor does not provide dedicated buttons for, including:
>
> - abbr
> - acronym
> - big
> - cite
> - code
> - q
> - samp
> - var

However, some of them aren't very common. For a start I'd suggest:

- abbr
- big and small
- cite
- code

**Comment by Graham-72** [comment]
I like this idea and would like to include at least one span style in order to show how to allow the user an opportunity to allow application of colour to words or phrases, e.g.
`span.l-blue|Light Blue`
`span.m-blue|Mid Blue`
with of course appropriate entries in the CSS files.

**Comment by stpaultim** [comment]
Giving this the "good first issue" tag with some hesitation. It does require a little knowledge of how CKEditor works, but I think this is fairly simple. A note to any first time contributor that there is likely to be much discussion about the right mix of tags to include, but providing an initial PR would get this issue moving.

**Comment by stpaultim** [comment]
I've provided a simple PR that may have a few problems, but I'll need feedback on it. 

My goal here is a simple improvement that does the following:

1) Makes the Styles widget functional when enabled so it doesn't appear broken. 
2) Provide some examples for anyone interested in adding their own custom styles to their theme or sub-theme of Basis. Configuring these custom styles can be very confusing. 

This PR does the following in the standard install profile:

1) It adds that three text color styles (Green, Blue, and Red) by default. They are not available to users until the Styles widget is added to a text filter. 
2) It adds the <span> tag as an acceptable tag in the HTML filter. (Needs discussion)
3) It adds this to basis.info = ckeditor_stylesheets[] = css/ckeditor-iframe.css
4) It adds css/ckeditor-iframe.css
5) It adds a couple of rules to css/skin.css

This PR will not effect existing sites. 
I avoided the tags suggested by @olafgrabienski, because they would require adding even more tags to the html filter (and we don't have this yet - https://github.com/backdrop/backdrop-issues/issues/2980).

Specific questions:

1) Is my approach reasonable. Am I putting my css in the right place?
2) What are some sensible Styles to include by default, within the constraints of current site config and functionality.

**Comment by klonos** [comment]
...hmm 🤔 ...Word offers these options (and it calls them "styles"):

![image](https://user-images.githubusercontent.com/2423362/137610616-91adf5fc-2252-4e33-a669-77f3e65d0114.png)

**Comment by klonos** [comment]
> Would it be possible to allow editing these formats via the admin UI, without ever having to touch code/files directly?

This is what I'm talking about: https://www.drupal.org/project/ckeditor_standalone_styles ...unfortunately, only D8/9 versions available 😓

**Comment by indigoxela** [comment]
There seems to be some confusion regarding "structure" and "styles". Adding special HTML tags and apply CSS are IMO different things.

Personally, I use CKEditor styles to let users apply predefined CSS classes to tags, without the need for them to know what "tag" or "class" means. Having an admin UI to provide these styles would be a cool thing.

The Ckeditor module in Backdrop has several hooks, so this might also be a good contrib candidate. I don't think, we can provide really useful styles OOTB, that fit every site.

**Comment by klonos** [comment]
> I don't think, we can provide really useful styles OOTB, that fit every site.

Yes, but the OOTB experience when adding the styles button is not ideal. What could help would be a "no styles defined yet" pseudo-element. Ideally, that element should be a link to online documentation.

In the meantime, providing a couple of styles OOTB that people can either remove, or tweak to their needs seems like a nice alternative. It's easier for people to do that rather than having to create new styles from scratch.

**Comment by stpaultim** [comment]
> The Ckeditor module in Backdrop has several hooks, so this might also be a good contrib candidate. I don't think, we can provide really useful styles OOTB, that fit every site.

I understand the difficulty of choosing useful styes OOTB that fit every site, but that is not the goal of this PR. The first line of the original issues states it pretty clearly:

"Description of the Need: Make it easier for new users to figure out how to edit the styles menu option in CKEditor."

I'm pretty good with Backdrop and I struggle with this every time I try to do it. Having some examples would have made it MUCH easier for me to configure my own. Also, it's seems really weird to me to provide a Styles menu/option without any styles under it. It looks broken to me.

**Comment by jenlampton** [comment]
We discussed this again in the design meeting today. Some things we noted:
* the examples we commit should not include color because it needs to work with all themes, and that can be really hard to do. (Color is in the current PR, but that is only an example and we expect to change it before commit)
* If we absolutely need to include color, maybe we should add the editor.css in the basis theme, and match the basis colors. Then there would be an example of the feature in use in core that people could copy, but people would not get that example with any other theme. 
* There are a few things we could do in core that would be theme agnostic, ideas include:
    - using "standard" classes that are added by core (things like `button-primary` and `button-secondary`). These would have already been overridden by most themes if they wanted different styles than the defaults. 
    - adding examples that relate to layout only (things like a `<ul>` that displays in 2 columns, 3 columns, or 4 columns)

**Comment by stpaultim** [comment]
@izmeez - Yes, in CKEditor 4 you have to edit your text editor and add the styles to the toolbar. 

Having said that, I'm not sure that this issiue is still relevent given our move to CKEditor 5. I'll need to do some testing to see how this works in CKEditor 5.

**Comment by indigoxela** [comment]
> I don't see the Styles option on the editing windows,

@izmeez  Yes, I know, that's weird UX. In order to be able to add styles, for example on admin/config/content/formats/filtered_html, you first have to drag the "Styles" button into the active toolbar. That's the same for CKE4 and CKE5 (in 5 it's in a vertical tab).

So, you can add styles via admin UI, but it's not obvious how.

**Comment by stpaultim** [comment]
The PR was still for CKEditor 4, so I'm sure that this needs work. I did some testing and I think the basic idea here may still make sense. But, it will need to be updated for CKEditor 5.
